### PR TITLE
Make brod dance with elvis

### PR DESCRIPTION
**Pull request by onno-vos-dev**
### Description
Introducing [elvis](https://github.com/inaka/elvis) to Brod to enforce style checks via Travis. Minor code changes needed to get Elvis to dance with Brod. A few rules or modules are excluded from Elvis since Elvis doesn't like to dance on those songs. 

### Open question
How do we want to run Elvis on Travis? There's a few options:
1. Compile Elvis on Travis. 
Advantage: Always got a clean Elvis binary built from source
Disadvantage: Time consuming
2. Build the Elvis escript and include it in the scripts repo. This way the only thing that has to be called from Travis is `elvis rock` and you'll be good to go.   
Advantage: fast. Takes around 2.5 - 3 seconds to run the elvis check
Disadvantage: "untrusted" binary is added to the repo. Updating the elvis binary becomes a possible security issue. Whoever creates a PR to update the binary could potentially sneak in arbitrary code. 

### Open tasks

- [ ] Possibly squash the commits before merging them in since the commits are a tad verbose. (Done on purpose to make reviewing easier).

### Commit log
commit b93732a42c5e826cda0c6306eb2da54e899e9760

    Elvis-fixes: Fix missing spaces

commit 8fc7c2ca12d2feabd2179d1c40b23cf2a5d20a33

    Elvis-fixes: Remove line-width-check. Elvis got this covered

commit 15dc3c83716c0c79629e6eb2e54606f45a0e420b

    Elvis-fixes: Fix used_ignored_variable

commit 754de4985d2971188bcaf2687898279c8bac5481

    Elvis-fixes: Use types when referring to records

commit 70dd752dc18d942eec0dfe0fae008b05f1dd52b1

    Elvis-fixes: Include elvis.config

    Couple exceptions are made compared to default elvis.config:
    - macro_names rule is turned off
    - dont_repeat_yourself has increaased min_complexity from 10 -> 15
    - brod_group_coordinator and brod_utils are excluded from nesting_level rule
    - brod and brod_utils are excluded from god_modules rule
    - brod_sock and brod_group_coordinator are excluded from invalid_dynamic_call rule

### CODE REVIEW

File: src/brod.erl [hunk 1]
```diff
@@ -156,6 +156,7 @@
                      | ?OFFSET_LATEST.
 -type message() :: kpro:message().
 -type error_code() :: kpro:error_code().
+-type kafka_message() :: #kafka_message{}.
```
Comment (zmstone) [inline src/brod.erl]: duplicated with `message()`, and please replace all below.

File: .travis.yml [hunk 2]
```diff
@@ -25,7 +25,6 @@ otp_release:
 script:
   - set -e
   - make vsn-check
-  - scripts/line-width-check.sh
```
Comment (zmstone) [inline .travis.yml:28]: please make travis install elvis and run check.

File: src/brod_utils.erl [hunk 3]
```diff
@@ -236,7 +236,7 @@ decode_messages(BeginOffset, Messages) when is_list(Messages) ->
 %% @doc Fetch a single message set from the given topic-partition.
 -spec fetch([endpoint()], topic(), partition(), offset(),
             non_neg_integer(), non_neg_integer(), pos_integer(),
-            sock_opts()) -> {ok, [#kafka_message{}]} | {error, any()}.
+            sock_opts()) -> {ok, [brod:kafka_message()]} | {error, any()}.
```
Comment (zmstone) [inline src/brod_utils.erl]: brod:message()

File: src/brod_topic_subscriber.erl [hunk 4]
```diff
@@ -76,7 +77,7 @@
 %%       partition-consumers are polling for more messages behind the scene
 %%       unless prefetch_count is set to 0 in consumer config.
 -callback handle_message(brod:partition(),
-                         #kafka_message{} | #kafka_message_set{},
+                         brod:kafka_message() | kafka_message_set(),
```
Comment (zmstone) [inline src/brod_topic_subscriber.erl]: brod:message()

File: src/brod_topic_subscriber.erl [hunk 5]
```diff
@@ -49,6 +49,7 @@
                       , cb_state()) -> cb_ret()).
 -type committed_offsets() :: [{brod:partition(), brod:offset()}].
 -type ack_ref()  :: {brod:partition(), brod:offset()}.
+-type kafka_message_set() :: #kafka_message_set{}.
```
Comment (zmstone) [inline src/brod_topic_subscriber.erl]: brod:message_set()

**Review by zmstone — COMMENTED** [review]
awesome.

**Comment by zmstone** [comment]
I vote to compile elvis in travis.

**Comment by onno-vos-dev** [comment]
- [X] Do not merge yet. There seems to be a problem with elvis on Travis. Will investigate more later this week. (See example: https://travis-ci.org/onno-vos-dev/brod/jobs/295992948 which should see elvis failing but it doesn't)

Edit: nevermind, I was an idiot and didn't commit my forcing failure. Here would be an example of Elvis no longer dancing: https://travis-ci.org/onno-vos-dev/brod/jobs/295997198

**Comment by onno-vos-dev** [comment]
@zmstone Fixed your comments and Travis now runs Elvis as well. I'm happy to squash as soon as you're done reviewing. :+1:

**Comment by zmstone** [comment]
very nice. squash please.

**Comment by zmstone** [comment]
btw, i'll take care of version bumping, changelog.md update etc in my on going pr when I rebase on top of this one.
will release in 3.3.0

**Review by zmstone — APPROVED** [review]

**Review by id — APPROVED** [review]
Good stuff!

**Comment by onno-vos-dev** [comment]
Thank you! 👍🏻

**Pull request merged**